client/src/components/squids/SquidList.jsx

```diff
@@ -2,12 +2,20 @@ import React from "react";
 
 import { useSquidList } from "../hooks/useSquidList";
 import { SquidListTile } from "./SquidListTile";
+import { SquidPagination } from "./SquidPagination";
 
 import "../../style/squids/squidList.pcss";
 
-export const SquidList = () => {
-  const { data, isLoading, isFetching, isError, error } = useSquidList();
-  const squids = data || [];
+export const SquidList = (props) => {
+  const {
+    location: { search },
+  } = props;
+
+  const pageNumber = parseInt(new URLSearchParams(search).get("pageNumber"), 10);
```

Comment:
🎉!

Reply:
If you found yourself reusing this logic elsewhere, this is a great use case for a custom hook -- useGetSearchParams, returning an object containing any/all search params at the current location